### ISSUE TITLE
Add approximation methods for missing states

### ISSUE THREAD

**Issue by mrlsdvd** [mrlsdvd/flight-price-prediction#1]
The simulation process used during Q-learning seems to be leaving out a lot of states. This may be due to the fact that the demand level is randomly assigned. As a result, the optimal action/price for a lot of states is $0.

We should figure out some approximation methods for these states.

Another thing we can try out is just passing through each possible demand level (as we do for the  number of days left and tickets left).

**Comment by minhanquinn** [comment]
Good idea! I think passing through each possible demand level could be good! Also note that a reason the optimal action for most states = 0 for ours is that our current version only does 1000 iterations, so it only sees max 1000 out of the 1,000,000 total states.

**Comment by mrlsdvd** [comment]
Good point. I ran the simulator for 10,000,000 iterations and the optimal action was still usually 0. Maybe there's another issue?
